ffi/diplomat/src/provider.rs

```diff
@@ -163,33 +183,47 @@ pub mod ffi {
             &mut self,
             other: &mut ICU4XDataProvider,
         ) -> DiplomatResult<(), ICU4XError> {
-            let a = core::mem::take(&mut self.0);
-            let b = core::mem::take(&mut other.0);
-            match (a, b) {
+            *self = match (core::mem::take(&mut self.0), core::mem::take(&mut other.0)) {
+                (ICU4XDataProviderInner::Empty, b) | (b, ICU4XDataProviderInner::Empty) => {
+                    ICU4XDataProvider(b)
+                }
+                (ICU4XDataProviderInner::Baked, ICU4XDataProviderInner::Baked) => {
+                    ICU4XDataProvider(ICU4XDataProviderInner::Baked)
+                }
+                #[cfg(feature = "any_provider")]
+                (ICU4XDataProviderInner::Any(a), ICU4XDataProviderInner::Baked) => {
```

Comment:
Thought/Issue: I think this means that if the caller invokes `fork_by_key`, even if they aren't using the baked provider, they get the massive Baked AnyProvider impl, with no way to disable it; even if it is an empty baked provider, that's still a lot of vtable and ZeroFrom

Reply:
Sure, but the `AnyProvider` itself will be the baked provider wrapped in locale fallback or forking, so there's no additional vtable.